assignments/arrays.js

```diff
@@ -63,34 +63,122 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
+function arrayListValueByKey(array,key) {
+  let returnArray = []
+  for (i = 0; i < array.length; i++) {
+    returnArray.push(array[i][key]);
+  };
+  return returnArray;
+}
+
+function describeCarByID(IDnum){
+  let returnArray = arrayListValueByKey(inventory,"id");
+  // this is just an error log in case something breaks in my code.
+  if (returnArray[0] === undefined) { console.log('Key array not extractable!')}
+  // confirmation log
+  else if (returnArray[0] !== false) { console.log('Key array extractable!'); };
+  // I now invoke the result...
+  // Now I invoke an indexOf on our use case and save it!
+  let carIndex = returnArray.indexOf(IDnum);
+  // Now I return a string describing the car at that index, so it can be console logged.
+  return 'Car ' + IDnum + ' is a ' + inventory[carIndex]["car_year"] + ' ' + inventory[carIndex]["car_make"] + ' ' + inventory[carIndex]["car_model"] + '.';
+};
+
+console.log(describeCarByID(33));
 
 
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
-let lastCar = 0;
-console.log();
+// define "last"? if it's the last added to the array (presumably the last one)...
+console.log(inventory.length);
+// gratuitous console.logs are just testing to see if this was working at all.
+let lastCar = inventory.length - 1;
+console.log(lastCar);
+console.log('The last car is a ' + inventory[lastCar]["car_make"] + ' ' + inventory[lastCar]["car_model"] + '.');
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
-let carModels = [];
-console.log();
+
+const carModels = arrayListValueByKey(inventory,"car_model");
+carModels.sort();
+// array joined in order to make it prettier and reduce the impact on my test window.
+console.log(carModels.join(' '));
 
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
-let carYears = [];
-console.log();
+
+// oh whoops, I already created this function! ^.~
+const carYears = arrayListValueByKey(inventory,"car_year");
+// array joined on log in order to make it prettier and reduce the impact on my test window. again.
+console.log(carYears.join(' '));
 
 // ==== Challenge 5 ====
 // The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
-let oldCars =[];
-console.log(); 
+
+// this is really quick with filter, right?
```

Comment:
you were not suppose to use filter for this because filter is a higher order function and those were suppose to not be used for this (however that might be changing so screw it, it's good with me).